massa-bootstrap/src/tests/binders.rs

```diff
@@ -632,3 +646,113 @@ fn test_client_drip_feed() {
+        server_keypair.clone(),
+        BootstrapSrvBindCfg {
+            max_bytes_read_write: 100,
+            thread_count: THREAD_COUNT,
+            max_datastore_key_length: MAX_DATASTORE_KEY_LENGTH,
+            randomness_size_bytes: BOOTSTRAP_RANDOMNESS_SIZE_BYTES,
+            consensus_bootstrap_part_size: CONSENSUS_BOOTSTRAP_PART_SIZE,
+            write_error_timeout: MassaTime::from_millis(1000),
+        },
+        Some(100),
+    );
+    let client_cfg = BootstrapClientBinder::test_default_config();
+    let mut client = BootstrapClientBinder::new(
+        client,
+        bootstrap_config.bootstrap_list[0].1.get_public_key(),
+        client_cfg,
+        Some(100),
+    );
+    let err_str = ['A'; 1_000].into_iter().collect::<String>();
+    let srv_err_str = err_str.clone();
+
+    let server_thread = std::thread::Builder::new()
+        .name("test_binders::server_thread".to_string())
+        .spawn({
+            move || {
+                let version: Version = Version::from_str("TEST.1.10").unwrap();
+
+                server.handshake_timeout(version, None).unwrap();
+
+                let before = Instant::now();
+                let message = server.next_timeout(None).unwrap();
+                match message {
+                    BootstrapClientMessage::BootstrapError { error } => {
+                        assert_eq!(error, srv_err_str);
+                    }
+                    _ => panic!("Bad message receive: Expected a peers list message"),
+                }
+                let dur = before.elapsed();
+                assert!(dur > Duration::from_secs(10));
+                assert!(dur < Duration::from_millis(11_500));
+
+                let before = Instant::now();
+                server
+                    .send_timeout(
+                        BootstrapServerMessage::BootstrapError { error: srv_err_str },
+                        None,
+                    )
+                    .unwrap();
+                let dur = before.elapsed();
+                assert!(dur > Duration::from_secs(10));
+                assert!(dur < Duration::from_millis(11_500));
+            }
+        })
+        .unwrap();
+
+    let client_thread = std::thread::Builder::new()
+        .name("test_binders::server_thread".to_string())
+        .spawn({
+            move || {
+                let version: Version = Version::from_str("TEST.1.10").unwrap();
+
+                client.handshake(version).unwrap();
+
+                let before = Instant::now();
+                client
+                    .send_timeout(
+                        &BootstrapClientMessage::BootstrapError {
+                            error: err_str.clone(),
+                        },
+                        None,
+                    )
+                    .unwrap();
+                let dur = before.elapsed();
+                assert!(dbg!(dur) > Duration::from_secs(10));
+                assert!(dur < Duration::from_millis(11_500));
```

Comment:
Please note that this kind of assert may fail sometimes

That's because TCP packets are bound to 64Kb per packet, so even if we allow more data to be sent, we'll have to do multiple read.
Even without any limiting (sleep) occuring, this means that it's possible to be still slower than the rate limit set.

Reply:
That could possibly explain why CI is failing on windows.

Any suggestions for limit setup in tests, and how much of an assert window to provide?

Reply:
yeah. Got things working with windows, but now mac is failing. takes about 20s. I'll account for mac, and assuming that's good, and green across the board, ill hit merge. Thanks!